scio-avro/src/main/scala/com/spotify/scio/avro/types/ConverterProvider.scala

```diff
@@ -139,11 +139,11 @@ private[types] object ConverterProvider {
 
     def cast(tree: Tree, tpe: Type): Tree = {
       tpe match {
-        case t if t =:= typeOf[Boolean] => tree
-        case t if t =:= typeOf[Int] => tree
-        case t if t =:= typeOf[Long] => tree
-        case t if t =:= typeOf[Float] => tree
-        case t if t =:= typeOf[Double] => tree
+        case t if t =:= typeOf[Boolean] => q"Boolean.box($tree)"
```

Comment:
Are these boxing really necessary? Shouldn't the compiler auto-box?

Reply:
Those are probably from here:
```scala
case (fieldName, value) =>
  q"if ($value != null) result.put($fieldName, $value)"
```

We can't get rid of the null check since `$value` can be null if it's not a primitive (String, nested record, etc.) but could never be null if it IS, i.e. `case class Record(a: Int, b: Float)`. So the warning is actually safe to ignore but maybe we can do something smarter there to get rid of the warning without adding boxing.